addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -113,13 +119,31 @@ def _compute_ongoing_sessions_count(self):
         "user_ids.livechat_ongoing_session_count",
     )
     def _compute_remaining_session_capacity(self):
+        agent_partner_ids = self.mapped('user_ids.partner_id').ids
+        count_by_agent_channel = {
+            (operator.id, channel.id): count
+            for operator, channel, count in self.env["discuss.channel"]._read_group(
+                [
+                    ("livechat_end_dt", "=", False),
+                    ("livechat_channel_id", "=", self.ids),
+                    ("livechat_operator_id", "in", agent_partner_ids),
```

Comment:
That's not correct: here we search, for every agent regardless of their channel only to filter out based on `channel.user_ids` later on. I think the domain should look like this to only incldues correct agents per channel.

```
    ("livechat_end_dt", "=", False) AND (
        (channel_id = A AND operator_id in A_agents) OR
        (channel_id = B AND operator_id in B_agents) OR
        ...
```